### PR TITLE
Fix ENVFS_RESOLVE_ALWAYS

### LINKED ISSUE #107: ENVFS_RESOLVE_ALWAYS=1 only works for entries in the fallback-path option
```
 ▶ ENVFS_RESOLVE_ALWAYS=1 ls -la /usr/bin/env
lr----x--t root root 0 B Thu Jan  1 01:00:00 1970  /usr/bin/env ⇒ /nix/store/mqvnklaf6mzmj7ng13aph5ymkvzpyhn0-fallback-path/env

 ▶ ENVFS_RESOLVE_ALWAYS=1 ls -la /usr/bin/systemd-detect-virt
lsd: /usr/bin/systemd-detect-virt: No such file or directory (os error 2).

 ▶ ENVFS_RESOLVE_ALWAYS=1 ls -la /usr/bin/bash
lsd: /usr/bin/bash: No such file or directory (os error 2).


 ▶ ENVFS_RESOLVE_ALWAYS=1 ls -la /usr/bin/cat
lsd: /usr/bin/cat: No such file or directory (os error 2).


 ▶ ENVFS_RESOLVE_ALWAYS=1 ls -la /usr/bin/sh
lr----x--t root root 0 B Thu Jan  1 01:00:00 1970  /usr/bin/sh ⇒ /nix/store/mqvnklaf6mzmj7ng13aph5ymkvzpyhn0-fallback-path/sh

 ▶ ls /nix/store/mqvnklaf6mzmj7ng13aph5ymkvzpyhn0-fallback-path
 env   login   mount   sh   wslpath
```

### PR DESCRIPTION
**Pull request by wentasah**
This, among other things, fixes #107. Additionally, I tried to resolve the FIXME comment about the open syscall, but I have no clue how to test that it actually works. The logic in the code didn't make much sense so I just made the code more sensible to me.

### CODE REVIEW

File: src/fs.rs [hunk 1]
```diff
@@ -270,11 +270,11 @@ where
         debug!("no syscall arguments received from /proc/<pid>/syscall");
         return None;
     }
-    // FIXME: We need to allow open/openat because some programs want to open themself, i.e. bash
-    let allowed_syscall = is_open_syscall(args[0])
-        && args[0] == libc::SYS_execve as usize
-        && !env.contains_key(OsStr::new("ENVFS_RESOLVE_ALWAYS"));
+    // We need to allow open/openat because some programs want to open themself, i.e. bash
+    let allowed_syscall =
+        is_open_syscall(args[0]) || env.contains_key(OsStr::new("ENVFS_RESOLVE_ALWAYS"));
```
Comment (Mic92) [inline src/fs.rs:275]: I am currently wondering, but shouldn't this also allow the execve systemcall?

Comment (wentasah) [reply src/fs.rs:275]: There is a special case for handling `execve` few lines below. This is only for handling the rest. I would prefer moving this part few lines below, close to the place where `allowed_syscall` is used. But I didn't want to make "too big" changes to the code.

Comment (Mic92) [reply src/fs.rs:275]: Check out https://github.com/Mic92/envfs/pull/146

Comment (Mic92) [reply src/fs.rs:275]: In the error case execve would still fall through.

**Comment by Mic92** [comment]
Applied!

**Pull request closed without merging**